src/react.ts

```diff
@@ -50,6 +50,56 @@ type Options = {
   sync?: boolean
 }
 
+/**
+ * useSnapshot
+ *
+ * Create a local snapshot that catches changes. Rule of thumb: read from snapshots, mutate the source.
+ * The component will only re-render when the parts of the state you access have changed, it is render-optimized.
+ *
+ * [Notes]
+ * Every object inside your proxy also becomes a proxy (if you don't use "ref"), so you can also use them to create
+ *   the local snapshot as seen on example B.
+ * Beware that you still can replace the child proxy with something else so it will break your snapshot.
```

Comment:
> can replace the child proxy

we can elaborate on this.

for example,
```js
> console.log(state)
{ profile: { name: "valtio" } }
> childState = state.profile
> console.log(childState)
{ name: "valtio" }
> state.profile.name = "react"
> console.log(childState)
{ name: "react" }
> state.profile = { name: "new name" }
> console.log(childState)
{ name: "react" }
> console.log(state)
{ profile: { name: "new name" } }
```

Reply:
I already liked how you wrote this so I just added it and reorganized the documentation.